Errors of imports in `*.d.ts*` files | Cannot find module 'x' or its corresponding type declarations. ts(2397)

**Describe the bug**
Every time I compile using `tsc` in the console, I get these errors, I guess it is just a problem with the paths in the `*.d.ts` files.

```
Errors  Files
     1  node_modules/custom-electron-titlebar/dist/index.d.ts:2
     3  node_modules/custom-electron-titlebar/dist/menu.d.ts:2
     1  node_modules/custom-electron-titlebar/dist/menubar.d.ts:2
     3  node_modules/custom-electron-titlebar/dist/menuitem.d.ts:2
     1  node_modules/custom-electron-titlebar/dist/titlebar.d.ts:2
```
**To Reproduce**
Steps to reproduce the behavior:
1. Run `tsc` in console
4. See error:

```
node_modules/custom-electron-titlebar/dist/index.d.ts:2:23 - error TS2307: Cannot find module 'vs/base/common/color' or its corresponding type declarations.

2 import { Color } from 'vs/base/common/color';
                        ~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menu.d.ts:2:23 - error TS2307: Cannot find module 'vs/base/common/color' or its corresponding type declarations.

2 import { Color } from "vs/base/common/color";
                        ~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menu.d.ts:4:28 - error TS2307: Cannot find module 'vs/base/common/lifecycle' or its corresponding type declarations.

4 import { Disposable } from "vs/base/common/lifecycle";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menu.d.ts:5:23 - error TS2307: Cannot find module 'vs/base/common/event' or its corresponding type declarations.

5 import { Event } from "vs/base/common/event";
                        ~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menubar.d.ts:2:23 - error TS2307: Cannot find module 'vs/base/common/event' or its corresponding type declarations.

2 import { Event } from 'vs/base/common/event';
                        ~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menuitem.d.ts:2:27 - error TS2307: Cannot find module 'vs/base/common/dom' or its corresponding type declarations.

2 import { EventLike } from "vs/base/common/dom";
                            ~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menuitem.d.ts:4:25 - error TS2307: Cannot find module 'vs/base/common/keyCodes' or its corresponding type declarations.

4 import { KeyCode } from "vs/base/common/keyCodes";
                          ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/menuitem.d.ts:5:28 - error TS2307: Cannot find module 'vs/base/common/lifecycle' or its corresponding type declarations.

5 import { Disposable } from "vs/base/common/lifecycle";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/custom-electron-titlebar/dist/titlebar.d.ts:2:23 - error TS2307: Cannot find module 'vs/base/common/color' or its corresponding type declarations.

2 import { Color } from 'vs/base/common/color';
                        ~~~~~~~~~~~~~~~~~~~~~~


Found 9 errors in 5 files.

Errors  Files
     1  node_modules/custom-electron-titlebar/dist/index.d.ts:2
     3  node_modules/custom-electron-titlebar/dist/menu.d.ts:2
     1  node_modules/custom-electron-titlebar/dist/menubar.d.ts:2
     3  node_modules/custom-electron-titlebar/dist/menuitem.d.ts:2
     1  node_modules/custom-electron-titlebar/dist/titlebar.d.ts:2
```

**Expected behavior**
No errors with imports in the console when compile is done.

**Screenshots**
![image](https://user-images.githubusercontent.com/45635827/193475084-95972b0b-49b5-44d0-bb60-53ca2881abcf.png)

**Desktop (please complete the following information):**
 - OS: Windows 11 Education
 - Electron version: 20.2.0
 - Node version: 16.16.0
 - Module version:  4.1.1
 - TypeScript version: 4.8.3

--- a/package.json
+++ b/package.json
@@ -59,6 +59,7 @@
     "ts-loader": "^9.4.1",
     "tsconfig-paths-webpack-plugin": "^4.0.0",
     "typescript": "^4.8.4",
+    "typescript-transform-paths": "^3.4.6",
     "webpack": "^5.74.0",
     "webpack-cli": "^4.10.0"
   }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -13,14 +13,6 @@
     "resolveJsonModule": true,
     "moduleResolution": "node",
     "strict": true,
-    "paths": {
-      "vs": [
-        "vs/*"
-      ],
-      "static": [
-        "static/*"
-      ],
-    },
     "skipLibCheck": true,
   },
   "include": [

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -1,5 +1,6 @@
 const path = require("path");
 const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
+const createTsTransformPaths = require('typescript-transform-paths').default;
 
 const srcPath = path.join(__dirname, "src");
 const distPath = path.join(__dirname, "dist");
@@ -14,7 +15,17 @@ const commonConfig = {
         rules: [
             {
                 test: /\.tsx?$/,
-                use: "ts-loader",
+                use: {
+                    loader: "ts-loader",
+                    options: {
+                        getCustomTransformers: (program) => ({
+                            before: [createTsTransformPaths(program, {})],
+                            afterDeclarations: [createTsTransformPaths(program, {
+                                afterDeclarations: true
+                            })]
+                        })
+                    }
+                },
                 exclude: /node_modules|\.d\.ts$/
             },
             {

